Monika After Story/game/script-ch30.rpy

```diff
@@ -826,6 +826,585 @@ init python:
+        """
+        # sync up current_monikatopic and eli data
+        # ALWAYS FIRST - this is core and important.
+        store.MASEventList.sync_current()
+
+
+    @ch30_reset(-960)
+    def xp():
+        """
+        Runs reset code specific for xp stuff
+        """
+        # xp fixes and adjustments
+        if persistent._mas_xp_lvl < 0:
+            persistent._mas_xp_lvl = 0 # prevent negative issues
+
+        if persistent._mas_xp_tnl < 0:
+            persistent._mas_xp_tnl = mas_xp.XP_LVL_RATE
+        elif int(persistent._mas_xp_tnl) > (2* int(mas_xp.XP_LVL_RATE)):
+            # likely time travel
+            persistent._mas_xp_tnl = 2 * mas_xp.XP_LVL_RATE
+
+        if persistent._mas_xp_hrx < 0:
+            persistent._mas_xp_hrx = 0.0
+
+        # setup initial xp values
+        mas_xp.set_xp_rate()
+        mas_xp.prev_grant = store.mas_getCurrSeshStart()
+
+
+    @ch30_reset(-940)
+    def name_eggs():
+        """
+        Runs reset code specific for name eggs
+        """
+        # name eggs
+        if mas_egg_manager.sayori_enabled() or store.mas_isO31():
+            mas_globals.show_sayori_lightning = True
+
+
+    @ch30_reset(-920)
+    def topic_lists():
+        """
+        Runs reset code specific for topic lists
+        """
+        # start by building event lists if they have not been built already
+        if not store.mas_events_built:
+            store.mas_rebuildEventLists()
+
+        # check if you've seen everything
+        if len(store.mas_rev_unseen) == 0:
+            # you've seen everything?! here, higher session limit
+            # NOTE: 1000 is arbitrary. Basically, endless monika topics
+            # I think we'll deal with this better once we hve a sleeping sprite
+            store.random_seen_limit = 1000
+
+
+    @ch30_reset(-900)
+    def rpy_file_check():
+        """
+        Runs reset code specific for the rpy file check
+        """
+        if not persistent._mas_pm_has_rpy:
+            if store.mas_hasRPYFiles():
+                if not store.mas_inEVL("monika_rpy_files"):
+                    store.queueEvent("monika_rpy_files")
+
+            else:
+                if persistent.current_monikatopic == "monika_rpy_files":
+                    persistent.current_monikatopic = 0
+                store.mas_rmallEVL("monika_rpy_files")
+
+
+    @ch30_reset(-880)
+    def games():
+        """
+        Runs reset code specific for games
+        """
+        # check for game unlocks
+        game_unlock_db = {
+            "chess": "mas_unlock_chess",
+            "hangman": "mas_unlock_hangman",
+            "piano": "mas_unlock_piano",
+            "nou": "mas_reaction_gift_noudeck",
+        }
+        store.mas_unlockGame("pong") # always unlock pong
+
+        for game_name, game_startlabel in game_unlock_db.iteritems():
+            # unlock if we've seen the label
+            if store.mas_getEVL_shown_count(game_startlabel) > 0:
```

Comment:
This won't unlock nou since `mas_reaction_gift_noudeck` isn't an event. Or maybe it's because we don't register reaction if you've given the game, so `mas_getEV` returns `None`.

Reply:
Maybe use `mas_seenEvent`? Altho it also checks current even list which might not be ideal.
Alternatively, add extra handling for nou like we have for pong.